### ISSUE TITLE
Add "Microsoft Graph Beta" API

### ISSUE THREAD

**Issue by baywet** [APIs-guru/openapi-directory#1053]
**Format**: OpenAPI 3.0
**Official**: YES
**Url**: https://graph.microsoft.com/beta
**Name**: Microsoft Graph Beta
**Category**: developer_tools
**Logo**: https://pbs.twimg.com/profile_images/1454912483248930822/_hO4WPRC_400x400.png

To be specific the description is present here but not in the index file. https://github.com/APIs-guru/openapi-directory/tree/main/APIs/microsoft.com/graph

**Comment by MikeRalphson** [comment]
Two API definitions, under `microsoft.com/graph` and `microsoft.com/graph-beta` should show up from tonight. Done like this to avoid clashes in version numbers (e.g. `1.0.1`).

**Comment by baywet** [comment]
Thanks. Is there a reason why you need to introduce an additional version number instead of relying on the version available in the path already?

**Comment by MikeRalphson** [comment]
The `info.version` field for both API definitions is `1.0.1` (unless I've done something wrong relinking the metadata) thus they would occupy the same point in our naming convention/directory-structure unless the "serviceName" differed. Does this help?

**Comment by baywet** [comment]
allow me to rephrase my question:
v1.0 and beta are already versions of the API. Could we use that instead of a semver version?

**Comment by MikeRalphson** [comment]
Our directory structure is `./APIs/{provider}/{service?}/{version}/openapi.yaml` where `{version}` comes from the OpenAPI `info.version` field, I think it might cause more confusion if we used 'v1.0` and `beta` as version numbers just for these APIs when people looking at the source of truth would be expecting `1.0.1`.

I'm open to further consideration.

**Comment by baywet** [comment]
Thanks the the additional context here. [The version specification](https://spec.openapis.org/oas/latest.html#fixed-fields-0) doesn't mandate any format.
My guess is we put that version as a placeholder because Microsoft graph is not versioned besides v1.0 or beta.
If we changed the original document to v1.0 and beta, would that be an issue for APIs guru?

**Comment by MikeRalphson** [comment]
No, that would be fine, and would allow us to put them both under a `graph` serviceName.

**Comment by baywet** [comment]
(created on issue on our side to address this)
Once we fix that, do you need a heads up or will the update "flow naturally" ?

**Comment by MikeRalphson** [comment]
It will involve a small manual step to 'merge' the serviceNames baack together, so would definitely appreciate a heads-up. Will keep this issue open in the meantime. Thanks.

**Comment by MikeRalphson** [comment]
@baywet - Closing due to inactivity - but please feel free to reopen the issue if necessary.

**Comment by baywet** [comment]
No worries. FYI people without contrib on the repo or so don't have the ability to re-open by themselves. We realized this the hard way a while ago. :)

**Comment by MikeRalphson** [comment]
You're right @baywet, I'll reword that canned response!

**Comment by baywet** [comment]
The version information is still incorrect because one PR was forgotten, so I authored this PR https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/989

@irvinesunday @peombwa do you think this could be a result of this change?  https://github.com/microsoft/OpenAPI.NET.OData/issues/324

**Comment by peombwa** [comment]
@baywet, @irvinesunday, the issue is due to OData function overloads having the same OperationIds. In this case, `/identityGovernance/entitlementManagement/assignments/additionalAccess()` and `/identityGovernance/entitlementManagement/assignments/additionalAccess(accessPackageId=''{accessPackageId}'',incompatibleAccessPackageId=''{incompatibleAccessPackageId}'')` share the same `operationId` - `identityGovernance.entitlementManagement.assignments.additionalAccess`.

OperationIds for the overloaded functions were missing in the OpenAPI document at https://github.com/APIs-guru/openapi-directory/tree/main/APIs/microsoft.com/graph, and now that they have been added, the duplicate is visible. The issue is closely related to https://github.com/microsoftgraph/msgraph-metadata/issues/289.

**Comment by baywet** [comment]
right, I had to come up with a "fancy" syntax when projecting those in Kiota (kiota doesn't rely on operation ids). Effectively the request builder for the parameter-less one is `additionaAccessRequestBuilder` and for the one with parameters it's `additionaAccessWithAccessPackageIdWithIncompatibleAccessPackageIdRequestBuilder`. Maybe we could do something similar for the operation ids?

**Comment by MikeRalphson** [comment]
Any update on this from our perspective, @baywet ?

**Comment by baywet** [comment]
according to spectral, this is still an issue at this point (duplicated operation id)
the version information is present though.

Let me create another issue to make sure it's properly addressed.

**Comment by MikeRalphson** [comment]
Thanks @baywet - subscribed.

**Comment by baywet** [comment]
@MikeRalphson thanks for your patience here.
We have resolved the duplicate Id issue and you should be able to process the descriptions now.
